Does not start with IPv6 disabled

Disabling IPv6 completely (using `ipv6.disable=1`) on the Linux command line) leads to xrdp not being able to start.

See https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=839266 for more details and logs!

Reproduced on CentOS 7.

1. Edit /etc/default/grub
2. Append "ipv6.disable=1" to GRUB_CMDLINE_LINUX
3. `# grub2-mkconfig -o /boot/grub2/grub.cfg`
4. Reboot
5. Edit X11RDP-RH-Matic.sh like following diff
6. `$ ./X11RDP-RH-Matic.sh --branch devel --nox11rdp`
7. `# systemctl start xrdp`
8. `# tail -f /var/log/xrdp.log`

~~~diff
diff --git a/X11RDP-RH-Matic.sh b/X11RDP-RH-Matic.sh
index 8e8b09d..fc39628 100755
--- a/X11RDP-RH-Matic.sh
+++ b/X11RDP-RH-Matic.sh
@@ -47,7 +47,7 @@ TARGETS="xrdp x11rdp"
 META_DEPENDS="rpm-build rpmdevtools"
 FETCH_DEPENDS="ca-certificates git wget"
 EXTRA_SOURCE="xrdp.init xrdp.sysconfig xrdp.logrotate xrdp-pam-auth.patch buildx_patch.diff x11_file_list.patch sesman.ini.master.patch sesman.ini.devel.patch"
-XRDP_CONFIGURE_ARGS="--enable-fuse --enable-jpeg --disable-static"
+XRDP_CONFIGURE_ARGS="--enable-fuse --enable-jpeg --disable-static --enable-ipv6"
 
 # flags
 PARALLELMAKE=true   # increase make jobs
~~~~

~~~
[20161012-17:47:57] [INFO ] starting xrdp with pid 2244
[20161012-17:47:57] [ERROR] xrdp_listen_main_loop: listen error, possible port already in use
~~~

--- a/common/os_calls.c
+++ b/common/os_calls.c
@@ -416,11 +416,27 @@ g_tcp_socket(void)
 
 #if defined(XRDP_ENABLE_IPV6)
     rv = (int)socket(AF_INET6, SOCK_STREAM, 0);
+    if (rv < 0)
+    {
+        log_message(LOG_LEVEL_ERROR, "g_tcp_socket: %s", g_get_strerror());
+
+        switch (errno)
+        {
+            case EAFNOSUPPORT: /* if IPv6 not supported, retry IPv4 */
+                log_message(LOG_LEVEL_INFO, "IPv6 not supported, falling back to IPv4");
+                rv = (int)socket(AF_INET, SOCK_STREAM, 0);
+                break;
+
+            default:
+                return -1;
+        }
+    }
 #else
     rv = (int)socket(AF_INET, SOCK_STREAM, 0);
 #endif
     if (rv < 0)
     {
+        log_message(LOG_LEVEL_ERROR, "g_tcp_socket: %s", g_get_strerror());
         return -1;
     }
 #if defined(XRDP_ENABLE_IPV6)

